AtReconstruction/AtFitter/OpenKF/kalman_filter/TrackFitterUKFTest.cxx

```diff
@@ -0,0 +1,268 @@
+   ASSERT_NEAR(m_ukf.matP()(2, 2), 0.15F, FLOAT_EPSILON);
+   ASSERT_NEAR(m_ukf.matP()(2, 3), 0.0F, FLOAT_EPSILON);
+
+   ASSERT_NEAR(m_ukf.matP()(3, 0), 0.0F, FLOAT_EPSILON);
+   ASSERT_NEAR(m_ukf.matP()(3, 1), 0.05F, FLOAT_EPSILON);
+   ASSERT_NEAR(m_ukf.matP()(3, 2), 0.0F, FLOAT_EPSILON);
+   ASSERT_NEAR(m_ukf.matP()(3, 3), 0.15F, FLOAT_EPSILON);
+
+   m_ukf.correctUKF(funcH, z);
+
+   // Expectations from the python results:
+   // ======================================
+   // x =
+   //     [ 2.4758845   0.53327217  0.21649734 -0.21214576]
+   // P =
+   // [[ 0.01433114 -0.01026142  0.00651178 -0.00465059]
+   //  [-0.01026142  0.0295458  -0.0046378   0.01344241]
+   //  [ 0.00651178 -0.0046378   0.13023154 -0.00210188]
+   //  [-0.00465059  0.01344241 -0.00210188  0.1333886 ]]
+
+   ASSERT_NEAR(m_ukf.vecX()[0], 2.4758845F, FLOAT_EPSILON);
+   ASSERT_NEAR(m_ukf.vecX()[1], 0.53327217F, FLOAT_EPSILON);
+   ASSERT_NEAR(m_ukf.vecX()[2], 0.21649734F, FLOAT_EPSILON);
+   ASSERT_NEAR(m_ukf.vecX()[3], -0.21214576F, FLOAT_EPSILON);
+
+   ASSERT_NEAR(m_ukf.matP()(0, 0), 0.01433114F, FLOAT_EPSILON);
+   ASSERT_NEAR(m_ukf.matP()(0, 1), -0.01026142F, FLOAT_EPSILON);
+   ASSERT_NEAR(m_ukf.matP()(0, 2), 0.00651178F, FLOAT_EPSILON);
+   ASSERT_NEAR(m_ukf.matP()(0, 3), -0.00465059F, FLOAT_EPSILON);
+
+   ASSERT_NEAR(m_ukf.matP()(1, 0), -0.01026142F, FLOAT_EPSILON);
+   ASSERT_NEAR(m_ukf.matP()(1, 1), 0.0295458F, FLOAT_EPSILON);
+   ASSERT_NEAR(m_ukf.matP()(1, 2), -0.0046378F, FLOAT_EPSILON);
+   ASSERT_NEAR(m_ukf.matP()(1, 3), 0.01344241F, FLOAT_EPSILON);
+
+   ASSERT_NEAR(m_ukf.matP()(2, 0), 0.00651178F, FLOAT_EPSILON);
+   ASSERT_NEAR(m_ukf.matP()(2, 1), -0.0046378F, FLOAT_EPSILON);
+   ASSERT_NEAR(m_ukf.matP()(2, 2), 0.13023154F, FLOAT_EPSILON);
+   ASSERT_NEAR(m_ukf.matP()(2, 3), -0.00210188F, FLOAT_EPSILON);
+
+   ASSERT_NEAR(m_ukf.matP()(3, 0), -0.00465059F, FLOAT_EPSILON);
+   ASSERT_NEAR(m_ukf.matP()(3, 1), 0.01344241F, FLOAT_EPSILON);
+   ASSERT_NEAR(m_ukf.matP()(3, 2), -0.00210188F, FLOAT_EPSILON);
+   ASSERT_NEAR(m_ukf.matP()(3, 3), 0.1333886F, FLOAT_EPSILON);
+}
+
+class TrackFitterUKFPhysicsTest : public testing::Test {
+public:
+   virtual void SetUp() override {}
+   virtual void TearDown() override {}
+
+   static constexpr float FLOAT_EPSILON{0.001F};
+
+   static constexpr size_t DIM_X{6};
+   static constexpr size_t DIM_V{2};
+   static constexpr size_t DIM_Z{3};
+   static constexpr size_t DIM_N{3};
+
+   kf::TrackFitterUKF<DIM_X, DIM_Z, DIM_V, DIM_N> m_ukf;
+
+   /// @brief to propagate the state vector using the process model
+   /// @param x state vector
+   /// @param v process noise vector
+   /// @return propagated (unaugmented) state vector
+   static kf::Vector<DIM_X> funcF(const kf::Vector<DIM_X> &x, const kf::Vector<DIM_V> &v)
+   {
+      // TODO: This needs to be filled with an RK4 solver for the physics model
+      kf::Vector<DIM_X> y{x};
+
+      // For now, we just return the state vector as is
+      return y;
+   }
```

Comment:
@Yassid This is the propagator. It takes in the state vector and noise vector as defined in the Overleaf.